addons/stock/models/stock_location.py

```diff
@@ -96,10 +96,23 @@ def default_get(self, fields):
     incoming_move_line_ids = fields.One2many('stock.move.line', 'location_dest_id') # used to compute weight
     net_weight = fields.Float('Net Weight', compute="_compute_weight")
     forecast_weight = fields.Float('Forecasted Weight', compute="_compute_weight")
+    empty_location = fields.Boolean('Empty Locations', compute="_compute_empty_location", readonly=True, store=True)
 
     _sql_constraints = [('barcode_company_uniq', 'unique (barcode,company_id)', 'The barcode for a location must be unique per company!'),
                         ('inventory_freq_nonneg', 'check(cyclic_inventory_frequency >= 0)', 'The inventory frequency (days) for a location must be non-negative')]
 
+    @api.depends('quant_ids.quantity', 'quant_ids.inventory_quantity')
+    def _compute_empty_location(self):
+        for location in self.filtered(lambda l: l.usage == 'internal'):
+            quantities = self.env['stock.quant'].read_group(
+                [('location_id', '=', location.id)],
+                ['quantity', 'inventory_quantity'],
+                ['location_id']
+            )
```

Comment:
try to use the read_group outside the loop. It's better for performance because it avoids doing a sql query by iteration